src/rosinstall_generator/generator.py

```diff
@@ -250,10 +256,11 @@ def generate_rosinstall(distro_name, names,
         names.update(include_names_from_path)
 
     # Allow special keywords in repos
-    repo_names, keywords = _split_special_keywords(repo_names or [])
-    if set(keywords).difference(set([ARG_ALL_PACKAGES])):
+    repo_names, repo_keywords = _split_special_keywords(repo_names or [])
+    if set(repo_keywords).difference(set([ARG_ALL_PACKAGES])):
         raise RuntimeError('The only keyword supported by repos is %r' % (ARG_ALL_PACKAGES))
 
+    keywords = _merge_two_keywords(pkg_keywords, repo_keywords)
```

Comment:
Merging both keywords vars looks awkward to me. Leaving it up to @sloretz to assess...